scipy/stats/tests/test_hypotests.py

```diff
@@ -159,3 +160,253 @@ def test_callable_cdf(self):
         r1 = cramervonmises(x, distributions.beta.cdf, args)
         r2 = cramervonmises(x, "beta", args)
         assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
+
+
+def assert_raises_with_match(exception_type, match, function, *args, **kwargs):
+    with assert_raises(exception_type, match=match):
+        function(*args, **kwargs)
+
+
+class TestSomersD(object):
+
+    def test_like_kendalltau(self):
```

Comment:
Not necessary to change now, but this is actually a large collection of tests.  It would be nice to split it into separate, focused tests.  (We don't have to copy the style of `test_kendalltau` *exactly*.)  But let's leave that for a future maintenance PR.